src/aws-lambda/recipes/recipes.py

```diff
@@ -1,13 +1,75 @@
 import json
+import re
+import boto3
+
+dynamodb = boto3.resource('dynamodb')
+ssm = boto3.client('ssm')
+
+response = ssm.get_parameter(Name='retaildemostore-recipes-table-name')
+ddbtable = dynamodb.Table(response['Parameter']['Value'])
+
+def getAllRecipes():
+    return ddbtable.scan()['Items']
+
+def getRecipeByIngredients(ingredient):
+    return ddbtable.get_item(
+        Key = { 
+            'PK' : 'INGREDIENTS#' + ingredient,
+            'SK' : 'INGREDIENTS#' + ingredient
+        }
+    )['Item']['recipes']
+    
+def getRecipeByCategory(category):
+    return {
+        "statusCode": 404,
+        "body": json.dumps('not implemented'  )
+    }
+    
+    
+def getRecipeByID(id):
+    return ddbtable.get_item(
+        Key = { 
+            'PK' : 'RECIPE#' + id,
+            'SK' : 'RECIPE#' + id
+        }
+    )['Item']
+
+def unsupported(path):
+        return {
+            "statusCode": 404,
+            "body": json.dumps('this request ' + path + ' is not supported by this API'  )
+        }
+
 
 def handler(event, _):
     http_method = event['httpMethod'].lower()
 
-    if http_method == 'get':
+    splitted_path = event['path'].split('/')
+    del(splitted_path[0]) # just ignore the first empty split as path starts with a /
+    json.dumps(splitted_path)
+    if splitted_path[0] != 'recipes':
         return {
-            "statusCode": 200,
-            "body": json.dumps('Recipes!')
+            "statusCode": 404,
+            "body": json.dumps('this request ' + requestparts[0] + ' is not supported by this API'  )
         }
+
+    if http_method == 'get':
+        if splitted_path[1] == 'id':
+            return {
+            "statusCode": 200,
+            "body": getRecipeByID(splitted_path[2])
+            }   
+        elif splitted_path[1] == 'ingredients':
+            return {
+            "statusCode": 200,
+            "body": getRecipeByIngredients(splitted_path[2])
```

Comment:
shouldn't we be using the event pathParameters if this is an API GW integration? also I think the ingredients need to be qs string params otherwise there's no scope to make the endpoint accept more than one ingredient in future